### ISSUE TITLE
Ripples doesn't consider transitive dependencies accurately

### ISSUE THREAD

**Issue by stephanenicolas** [autonomousapps/dependency-analysis-android-gradle-plugin#334]
In `0.68.0` the `:ripples` task will output things like:

```
- Remove <dependency> from 'api'
    :project:a uses this dependency transitively. You should add it to 'api'
    :project:b uses this dependency transitively. You should add it to 'implementation'
   ...
```

There are some issues with this approach: 
* `project a` could be using a transitive dependency of `<dependency>` and not `<dependency>` itself.
* there is no way to output this right now
* I believe there is no real "bad advice" being given as I never faced a case that doesn't compile. But it is highly possible that the advice is too large and should be more accurate, recommending to add one of the dependencies of `<dependency>` instead of  `<dependency>` itself.

**Comment by autonomousapps** [comment]
As I said elsewhere, I think there might be a more general issue with how the plugin handles transitivity. I had a bit of an inspiration this morning, though, so hopefully I can resolve this sooner than later.

**Comment by stephanenicolas** [comment]
in `0.69.1-SNAPSHOT`, 

We have 
`A uses annotation --- depends on --> B ---depends on --> C defines annotation`
following an advice in `B` , I downgrade `api project(C)` to `implementation project(C)` 
but the ripple in `project(A)` is not detected and `A` doesn't compile anymore.

**Comment by stephanenicolas** [comment]
In an other case, it is a transitive dependency that is not taken into account:

We have
`A  --- depends on --> B ---depends on --> C --- api depends on --> D`
following an advice in B , I downgrade `api project(C) to implementation project(C)`
but the ripple in project(A) is not detected and A doesn't compile anymore. `A` is using `D` which was obtained indirectly.

**Comment by autonomousapps** [comment]
This was recently resolved in 0.71.0.
